### PR TITLE
Use preventProjectMigrations property to prevent migrations (Sister PR in local-dev-lib)

### PR DESCRIPTION
**Pull request by kemmerle**
## Description and Context
Chris C. added an endpoint that returns metadata for a single public app. He also added a new property, `preventProjectMigrations`, a boolean that returns true when the app is a marketplace app that has not been added to the Ecosystem Quality review whitelist. This ties in disabling project migrations directly with the allow-list, which better future proofs this command.  

The sister [PR](https://github.com/HubSpot/hubspot-local-dev-lib/pull/157) in local-dev-lib has been merged, and the local-dev-lib version is updated. 

## Screenshots

Marketplace apps not on the whitelist are disabled in the dropdown menu:

Screenshot 2024-07-11 at 11 43 58 AM

A similar error is returned with an `appId` flag:

Screenshot 2024-07-11 at 11 44 53 AM

@robrown-hubspot, Laurie Q., Sandra C., and I discussed whether to make the error messages more detailed, but we decided to keep the errors generalized and point developers to internal documentation for more info. 

## TODO

- [ ] Address feedback 

## Who to Notify

@brandenrodgers @joe-yeager

### CODE REVIEW

File: packages/cli/commands/project/migrateApp.js
Comment (brandenrodgers) [inline packages/cli/commands/project/migrateApp.js:85]: This is a pretty massive try catch block. It might make our lives easier down the road if you break this up into smaller, more focused blocks. My assumption is that you moved this up here to include the call to `fetchPublicAppMetadata `.

File: packages/cli/commands/project/migrateApp.js
Comment (brandenrodgers) [inline packages/cli/commands/project/migrateApp.js:87]: Is it possible for apps to not have a name?

**Comment by brandenrodgers** [comment]
The code in here lgtm, but I haven't tested it out yet. I'll hold off on testing until the local-dev-lib changes are merged and released

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js:85]: That's right--I can definitely break this up!

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js:87]: I think apps must have names. Let me check this again, because I think I did this to avoid a TypeScript error.

Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js]: Addressed in https://github.com/HubSpot/hubspot-cli/pull/1111/commits/47aefe6e63ac137ba4ec03a9acb7668c509a27b6

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js:85]: @brandenrodgers, I'm having a little trouble breaking up the try...catch block for two reasons:

1) I need access to the `selectedApp` (ln. 86), so I can display the warning later (ln. 122). 
2) I need access to `projectLocation` (ln. 100) when downloading the zipped project files (ln. 163). 

Any advice on this?

Comment (brandenrodgers) [reply packages/cli/commands/project/migrateApp.js:85]: If I'm understanding correctly, you should be able to declare the variables outside the scope of the try/catch block (i.e. use `let`)

Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js:85]: Oof just failed scoping 🤦‍♀️

**Comment by kemmerle** [comment]
Feedback from Slack:

- [ ] When no apps can be migrated, do not display the dropdown menu.
- [ ] Display more generic disabled messages when apps can't be migrated (don't specify marketplace app).

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js:85]: Addressed in https://github.com/HubSpot/hubspot-cli/pull/1111/commits/b556b9c5946602db821e510edb18917e57231b76

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js]: Got too fancy. Addressed in https://github.com/HubSpot/hubspot-cli/pull/1111/commits/9f5e14cfa90c915798885436b2505a45e33e514d

File: packages/cli/commands/project/migrateApp.js
Comment (kemmerle) [reply packages/cli/commands/project/migrateApp.js]: Moved `appName` below; addressed in https://github.com/HubSpot/hubspot-cli/pull/1111/commits/9f5e14cfa90c915798885436b2505a45e33e514d

**Review by camden11 — APPROVED** [review]

**Pull request merged**
